web/src/main/java/com/google/accompanist/web/WebView.kt

```diff
@@ -80,45 +88,120 @@ fun WebView(
+ * wrapper as an example.
+ *
+ * The WebView attempts to set the layoutParams based on the Compose modifier passed in. If it
+ * is incorrectly sizing, use the layoutParams composable function instead.
+ *
+ * @param state The webview state holder where the Uri to load is defined.
+ * @param layoutParams A FrameLayout.LayoutParams object to custom size the underlying WebView.
+ * @param modifier A compose modifier
+ * @param captureBackPresses Set to true to have this Composable capture back presses and navigate
+ * the WebView back.
+ * @param navigator An optional navigator object that can be used to control the WebView's
+ * navigation from outside the composable.
+ * @param onCreated Called when the WebView is first created, this can be used to set additional
+ * settings on the WebView. WebChromeClient and WebViewClient should not be set here as they will be
+ * subsequently overwritten after this lambda is called.
+ * @param onDispose Called when the WebView is destroyed. Provides a bundle which can be saved
+ * if you need to save and restore state in this WebView.
+ * @param client Provides access to WebViewClient via subclassing
+ * @param chromeClient Provides access to WebChromeClient via subclassing
+ * @param factory An optional WebView factory for using a custom subclass of WebView
+ */
+@Composable
+fun WebView(
+    state: WebViewState,
+    layoutParams: FrameLayout.LayoutParams,
+    modifier: Modifier = Modifier,
+    captureBackPresses: Boolean = true,
+    navigator: WebViewNavigator = rememberWebViewNavigator(),
+    onCreated: (WebView) -> Unit = {},
+    onDispose: (WebView) -> Unit = {},
+    client: AccompanistWebViewClient = remember { AccompanistWebViewClient() },
+    chromeClient: AccompanistWebChromeClient = remember { AccompanistWebChromeClient() },
+    factory: ((Context) -> WebView)? = null,
+) {
+    val webView = state.webView
 
-        snapshotFlow { state.content }.collect { content ->
-            when (content) {
-                is WebContent.Url -> {
-                    webView?.loadUrl(content.url, content.additionalHttpHeaders)
-                }
+    BackHandler(captureBackPresses && navigator.canGoBack) {
+        webView?.goBack()
+    }
 
-                is WebContent.Data -> {
-                    webView?.loadDataWithBaseURL(
-                        content.baseUrl,
-                        content.data,
-                        content.mimeType,
-                        content.encoding,
-                        content.historyUrl
-                    )
-                }
+    webView?.let { wv ->
+        LaunchedEffect(wv, navigator) {
+            with(navigator) {
+                wv.handleNavigationEvents()
             }
         }
-    }
 
-    val currentOnDispose by rememberUpdatedState(onDispose)
+        LaunchedEffect(wv, state) {
+            state.viewState?.let {
+                wv.restoreState(it)
```

Comment:
Is it possible to call this as part of the `AndroidView` factory? As is, I think we will have about a 2-composition delay before this is called:

- First composition, `webView` is `null`
- Second composition, `LaunchedEffect` is called
- After second composition, this line will run

Reply:
Great idea, the current place felt wrong to me as well